chartfx-dataset/src/test/java/de/gsi/dataset/spi/DoubleDataSetTests.java

```diff
@@ -54,6 +61,65 @@ public void defaultTests() {
         assertEquals(0, firstDataSet.getDataCount(), "check clear method");
     }
 
+    public static void checkAddPointsAtStart(final DoubleDataSet dataSet, final int testCase) {
+        final String dsType = dataSet.getClass().getSimpleName();
+
+        final int nDim = dataSet.getDimension();
+        final int nData = dataSet.getDataCount();
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.atDebug().addArgument(dsType).addArgument(nDim).addArgument(nData).log("info: data set '{}' with nDim = {} and nData = {}");
+        }
+
+        for (int i = 0; i < testCoordinate[0].length; i++) {
+            DoubleDataSet ret;
+
+            if (testCase == 0) {
+                // X & Y coordinates only
+                ret = dataSet.add(0, testCoordinate[0][i], testCoordinate[1][i]);
+                assertEquals(dataSet, ret, "check '" + dsType + "' return value (fluent design)");
+                assertEquals(nData + i + 1, dataSet.getDataCount(), "check '" + dsType + "' data point count");
+            } else if (testCase == 1) {
+                // X, Y, and label
+                ret = dataSet.add(0, testCoordinate[0][i], testCoordinate[1][i], "label" + i);
+                assertEquals(dataSet, ret, "check '" + dsType + "' return value (fluent design)");
+                assertEquals(nData + i + 1, dataSet.getDataCount(), "check '" + dsType + "' data point count");
+            } else if (testCase == 2) {
+                // X, Y (via arrays and at the back) but w/o label
+                ret = dataSet.add(0, testCoordinate[0][i], testCoordinate[1][i]);
+                assertEquals(dataSet, ret, "check '" + dsType + "' return value (fluent design)");
+                assertEquals(nData + i + 1, dataSet.getDataCount(), "check '" + dsType + "' data point count");
+            } else if (testCase == 3) {
+                // X, Y, error coordinates (via arrays) but w/o label
+                if (i == 0) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)